src/main/java/seedu/address/model/person/LoanRate.java

```diff
@@ -0,0 +1,58 @@
+package seedu.address.model.person;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Loan's rate in the loan book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidRate}
+ */
+public class LoanRate {
+
+    public static final String LOANRATE_VALIDATION_REGEX = "([1-9]\\d*(\\.\\d*[1-9])?)|(0\\.\\d*[1-9])";
+    public static final String MESSAGE_LOANRATE_CONSTRAINTS =
+            "Rate should be greater than 0 and have at most 2 decimal places. The last decimal place cannot be 0.";
+
+    public final double rate;
+
+    /**
+     * Constructs a {@code LoanRate}.
+     *
+     * @param rate A valid rate.
+     */
+    LoanRate(double rate) {
+        if (Math.round(rate) - rate == 0) {
+            checkArgument(isValidRate(String.valueOf((int) rate)), MESSAGE_LOANRATE_CONSTRAINTS);
+        } else {
+            checkArgument(isValidRate(String.valueOf(rate)), MESSAGE_LOANRATE_CONSTRAINTS);
+        }
+        this.rate = rate;
+    }
```

Comment:
While it is okay to have a constructor that directly initializes the LoanRate value, the main value of having a constructor that takes in a string is due to the fact that the command parser will pass a string (the user's input) here.
In short, please make a contructor that accepts a string as a parameter, not a double.
`Double.parseDouble()` can help do the heavy lifting for you.
You can also refer to [`Double.valueOf​(String s)`](https://docs.oracle.com/javase/9/docs/api/java/lang/Double.html#valueOf-java.lang.String-) for the regex of a double value, but I'll suggest catching the `NumberFormatException` thrown by `Double.parseDouble()` for the validity check instead.